source/containers.tex

```diff
@@ -20957,3 +21023,631 @@
+\end{note}
+
+\item
+otherwise,
+\tcode{submdspan_mapping_result\{layout_stride::mapping(sub_ext, sub_strides),\linebreak{}offset\}}.
+\end{itemize}
+\end{itemdescr}
+
+\rSec4[mdspan.submdspan.submdspan]{\tcode{submdspan} template function}
+
+\indexlibraryglobal{submdspan}%
+\begin{itemdecl}
+template<class ElementType, class Extents, class LayoutPolicy,
+         class AccessorPolicy, class... SliceSpecifiers>
+  constexpr auto submdspan(
+    const mdspan<ElementType, Extents, LayoutPolicy, AccessorPolicy>& src,
+    SliceSpecifiers... slices) -> @\seebelow@;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{index_type} be \tcode{typename Extents::index_type}.
+
+\pnum
+Let \tcode{sub_map_offset} be the result of
+\tcode{submdspan_mapping(src.mapping(), slices...)}.
+\begin{note}
+This invocation of \tcode{submdspan_mapping}
+selects a function call via overload resolution
+on a candidate set that includes the lookup set
+found by argument dependent lookup\iref{basic.lookup.argdep}.
+\end{note}
+
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{sizeof...(slices)} equals \tcode{Extents::rank()}, and
+\item
+the expression \tcode{submdspan_mapping(src.mapping(), slices...)}
+is well-formed when treated as an unevaluated operand.
+\end{itemize}
+
+\pnum
+\mandates
+\begin{itemize}
+\item
+\tcode{decltype(submdspan_mapping(src.mapping(), slices...))}
+is a specialization of \tcode{submd-\linebreak{}span_mapping_result}.
+
+\item
+\tcode{is_same_v<remove_cvref_t<decltype(sub_map_offset.mapping.extents())>,}
+\tcode{decltype(\linebreak{}submdspan_extents(src.mapping(), slices...))>}
+is \tcode{true}.
+
+\item
+For each rank index $k$ of \tcode{src.extents()},
+exactly one of the following is true:
+  \begin{itemize}
+  \item $S_k$ models \tcode{\libconcept{convertible_to}<index_type>},
+  \item $S_k$ models \tcode{\exposconcept{index-pair-like}<index_type>},
+  \item \tcode{is_convertible_v<$S_k$, full_extent_t>} is \tcode{true}, or
+  \item $S_k$ is a specialization of \tcode{strided_slice}.
+  \end{itemize}
+\end{itemize}
+
+\pnum
+\expects
+\begin{itemize}
+\item
+For each rank index $k$ of \tcode{src.extents()},
+all of the following are \tcode{true}:
+  \begin{itemize}
+  \item
+  if $S_k$ is a specialization of \tcode{strided_slice}
+    \begin{itemize}
+    \item $\tcode{$s_k$.extent} = 0$, or
+    \item $\tcode{$s_k$.stride} > 0$
+    \end{itemize}
+  \item
+  $0 \le \tcode{\exposid{first_}<index_type, $k$>(slices...)}$
+  $\le \tcode{\exposid{last_}<$k$>(src.extents(), slices...)}$
+  $\le \tcode{\linebreak{}src.extent($k$)}$
+  \end{itemize}
+
+\item
+\tcode{sub_map_offset.mapping.extents() == submdspan_extents(src.mapping(), slices...)}\linebreak
+is \tcode{true}; and
+
+\item
+for each integer pack \tcode{I} which is a multidimensional index
+in \tcode{sub_map_offset.mapping.extents()},
+\begin{codeblock}
+sub_map_offset.mapping(I...) + sub_map_offset.offset ==
+  src.mapping()(@\exposid{src-indices}@(array{I...}, slices ...))
```

Comment:
Missing part of <https://github.com/cplusplus/draft/pull/6338#discussion_r1240037371>.
```suggestion
  src.mapping()(@\exposid{src-indices}@(array{I...}, slices...))
```

Reply:
Many thanks, fixed.